lib/common/socket.c

```diff
@@ -798,12 +800,16 @@ static int on_async_resumption_new(SSL *ssl, SSL_SESSION *session)
     i2d_SSL_SESSION(session, &p);
 
     id = SSL_SESSION_get_id(session, &id_len);
-    resumption_new(h2o_iovec_init(id, id_len), data);
+    resumption_new(sock, h2o_iovec_init(id, id_len), data);
     return 0;
 }
 
 static void on_async_resumption_remove(SSL_CTX *ssl_ctx, SSL_SESSION *session)
 {
+    if (resumption_remove == NULL) {
+        return;
+    }
```

Comment:
Thank you for the clarification. Looking at the man page, it says:
> The remove_session_cb() is called, whenever the SSL engine removes a session from the internal cache. This happens
> * when the session is removed because it is expired
> * or when a connection was not shutdown cleanly.
> * It also happens for all sessions in the internal session cache when SSL_CTX_free is called.
> (source: https://www.openssl.org/docs/man1.0.1/ssl/SSL_CTX_sess_set_remove_cb.html)

So it would be reasonable for an _external_ cache (i.e. a cache that uses memcached or redis) to do nothing when the event gets fired (note also that we would not want to remove a cache entry when a connection is abruptly shut down, which is often the case with the browsers).

To summarize, I support making this change.

OTOH, I do oppose to making this change within this PR. Changing how H2O responds to the remove callback is a bug fix. Therefore, it should go through a separate PR to master, especially since we might want to backport this to 2.0 branch as well.

I would appreciate it if you create a PR against master that fixes the issue. The PR should contain sufficient information describing why the code need to be changed such way.